### ISSUE TITLE
[Feature request] Add support for removing unnecessary data from links

### ISSUE THREAD

**Issue by DerekTurtleRoe** [podaboutlist/linkfix-for-discord#70]
Some examples could include:

- YouTube (remove things like si=, etc)
- Twitter (remove things like t=)
- Amazon (remove all the extra fluff they add to sharable links)
etc

This could also be customized via commands for services that aren't supported yet, like:

Add site: /addsite example.com/tracking=referrer
Replace with: /replacewith example.com

or something like that. This + #4 and #7 would make it pretty much perfect. 👍 

TL;DR it would be nice if we could add support for removing tracking and other useless or unnecessary info from links, and have custom site support. 😄

**Comment by RalphORama** [comment]
Thanks for the suggestion!

LinkFix already removes tracking stuff when fixing YouTube Shorts and Twitter links (see [here](https://github.com/podaboutlist/linkfix-for-discord/blob/main/src/replacements/BaseReplacement.ts#L61)). I'm still working on database implementation for toggling features but once I get that done I'd be happy to add support for stripping regular YouTube links, Amazon, eBay, etc.

The goal is to be "set and forget" so I don't want to add features to modify links we aren't strictly "fixing" (i.e. providing a URL that better embeds, `x.com` -> `fixupx.com`) until I have the code implemented to toggle that kind of thing on/off on a per-server basis.

**Comment by DerekTurtleRoe** [comment]
@RalphORama If it isn't difficult I can try to look into adding those removal additions after you have added per-server configuration.
